File: 2130-maximum-twin-sum-of-a-linked-list/2130-maximum-twin-sum-of-a-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    int pairSum(ListNode* head) {
        ListNode *slow = head, *fast = head;
        while(fast && fast->next){
            slow = slow->next;
            fast = fast->next->next;
        }
        ListNode *nxt, *prev = NULL;
        while(slow != NULL){
            nxt = slow->next;
            slow->next = prev;
            prev = slow;
            slow = nxt;
        }
        int ans = 0;
        ListNode *start = head;
        while(prev != NULL){
            ans = max(ans, prev->val+start->val);
            prev = prev->next;
            start = start->next;
        }
        return ans;
    }
};

File: 2130-maximum-twin-sum-of-a-linked-list/2130-maximum-twin-sum-of-a-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def pairSum(self, head: Optional[ListNode]) -> int:
        slow, fast = head, head
        while fast and fast.next:
            slow = slow.next
            fast = fast.next.next
        nxt, prev = None, None
        while slow:
            nxt = slow.next
            slow.next = prev
            prev = slow
            slow = nxt
        start = head
        ans = 0
        while prev:
            ans = max(ans, start.val+prev.val)
            start = start.next
            prev = prev.next
        return ans